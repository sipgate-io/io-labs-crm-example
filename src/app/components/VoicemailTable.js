import React from 'react';
import PropTypes from 'prop-types';
import './VoicemailTable.css';

export const VoicemailTable = ({ voicemails, deleteVoicemail, hideText }) => {
    const renderContent = () => {
        if (voicemails.length > 0) {
            return voicemails.map((voicemail, index) => {
                return (
                    <tr key={index}>
                        <td>{voicemail.number}</td>
                        <td className="tableText">
                            <span>
                                {voicemail.showText
                                    ? voicemail.text
                                    : 'Text is hidden'}
                            </span>
                            <button
                                className="showHideButton"
                                onClick={() => hideText(index)}
                            >
                                {voicemail.showText ? 'hide' : 'show'}
                            </button>
                        </td>
                        <td>{voicemail.duration} seconds</td>
                        <td>
                            {
                                <button
                                    onClick={() => deleteVoicemail(index)}
                                    className="deleteButton"
                                >
                                    remove
                                </button>
                            }
                        </td>
                    </tr>
                );
            });
        }

        return (
            <tr>
                <td colSpan={3}>There is no entry</td>
            </tr>
        );
    };

    return (
        <div className="voicemailContainer">
            <h2>Voicemails</h2>
            <table className="voicemailTable">
                <colgroup>
                    <col className="numberColumn" />
                    <col />
                    <col className="durationColumn" />
                    <col className="removeColumn" />
                </colgroup>
                <tr>
                    <th>Number</th>
                    <th>Text</th>
                    <th>Duration</th>
                    <th></th>
                </tr>
                {renderContent()}
            </table>
        </div>
    );
};

VoicemailTable.propTypes = {
    voicemails: PropTypes.array,
    deleteVoicemail: PropTypes.func,
    hideText: PropTypes.func,
};
