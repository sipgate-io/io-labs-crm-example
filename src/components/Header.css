header {
    width: 100%;
}

#headerContent {
    margin-top: 2em;
    text-align: center;
}
