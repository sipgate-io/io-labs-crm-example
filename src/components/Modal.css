.container {
    position: fixed;
    top: 0;
    left: 0;
    height: 100vh;
    width: 100vw;
    background-color: rgba(0, 0, 0, 0.7);
    --vibrating-intensity: 0.1em;
}

.modal {
    display: flex;
    align-items: center;
    height: 80%;
    justify-content: center;
}
.modal-inner {
    text-align: center;
    flex: 1;
    max-width: 50%;
    background-color: white;
    padding: 1em;
    border: 1px solid transparent;
    border-radius: 25px;
}

.modal-vibrating {
    animation: modal-inner-icon 3000ms infinite;
}

@keyframes modal-inner-icon {
    0% {
        transform: translate3d(0em, 0, 0);
    }
    2% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    4% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    6% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    8% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    10% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    12% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    14% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    16% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    18% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    20% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    22% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    24% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    26% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    28% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    30% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    32% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    34% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    36% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    38% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    40% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    42% {
        transform: translate3d(var(--vibrating-intensity), 0, 0);
    }
    44% {
        transform: translate3d(calc(var(--vibrating-intensity) * -1), 0, 0);
    }
    46% {
        transform: translate3d(0em, 0, 0);
    }
}
