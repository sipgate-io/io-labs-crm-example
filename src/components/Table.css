.voicemailTable {
    margin-left: auto;
    margin-right: auto;
}

.voicemailContainer {
    text-align: center;
}

td {
    padding: 0px 20px;

}