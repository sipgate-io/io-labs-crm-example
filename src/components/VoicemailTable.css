.voicemailTable {
    margin-left: auto;
    margin-right: auto;
}

.voicemailContainer {
    text-align: center;
}

table {
    border-collapse: collapse;
    width: 600px;
}

th, td {
    padding: 5px 20px;
}

tr:nth-child(even) {
    background-color: lightgray;
}

th {
    background-color: black;
    color: white;
}