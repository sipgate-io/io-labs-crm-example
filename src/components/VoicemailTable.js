import React from 'react';
import PropTypes from 'prop-types';
import './VoicemailTable.css';

export const VoicemailTable = ({ voicemails, deleteVoicemail }) => {
    const renderContent = () => {
        if (voicemails.length > 0) {
            return voicemails.map((voicemail, index) => {
                return (
                    <tr key={index}>
                        <td>{voicemail.number}</td>
                        <td>{voicemail.text}</td>
                        <td>{voicemail.duration} seconds</td>
                        <td>
                            {
                                <button onClick={() => deleteVoicemail(index)}>
                                    remove
                                </button>
                            }
                        </td>
                    </tr>
                );
            });
        }

        return (
            <tr>
                <td colSpan={3}>There is no entry</td>
            </tr>
        );
    };

    return (
        <div className="voicemailContainer">
            <h2>Voicemails</h2>
            <table className="voicemailTable">
                <colgroup>
                    <col className="numberColumn" />
                    <col />
                    <col className="durationColumn" />
                    <col className="removeColumn" />
                </colgroup>
                <tr>
                    <th>Number</th>
                    <th>Text</th>
                    <th>Duration</th>
                    <th></th>
                </tr>
                {renderContent()}
            </table>
        </div>
    );
};

VoicemailTable.propTypes = {
    voicemails: PropTypes.array,
    deleteVoicemail: PropTypes.func,
};
