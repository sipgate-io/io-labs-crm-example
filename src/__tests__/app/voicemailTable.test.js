import renderer from 'react-test-renderer';
import React from 'react';
import { VoicemailTable } from '../../app/components/VoicemailTable';

it('it renders VoicemailTable without data', () => {
    const tree = renderer
        .create(
            <VoicemailTable
                voicemails={[]}
                deleteVoicemail={() => {}}
                hideText={() => {}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('it renders VoicemailTable with example data', () => {
    const tree = renderer
        .create(
            <VoicemailTable
                voicemails={[
                    { number: 'unknown', text: 'Test 1', duration: 25 },
                ]}
                deleteVoicemail={() => {}}
                hideText={() => {}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
