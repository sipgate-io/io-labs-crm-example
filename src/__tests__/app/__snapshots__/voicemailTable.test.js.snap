// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders VoicemailTable with example data 1`] = `
<div
  className="voicemailContainer"
>
  <h2>
    Voicemails
  </h2>
  <table
    className="voicemailTable"
  >
    <colgroup>
      <col
        className="numberColumn"
      />
      <col />
      <col
        className="durationColumn"
      />
      <col
        className="removeColumn"
      />
    </colgroup>
    <tr>
      <th>
        Number
      </th>
      <th>
        Text
      </th>
      <th>
        Duration
      </th>
      <th />
    </tr>
    <tr>
      <td>
        unknown
      </td>
      <td
        className="tableText"
      >
        <span>
          Text is hidden
        </span>
        <button
          className="showHideButton"
          onClick={[Function]}
        >
          show
        </button>
      </td>
      <td>
        25
         seconds
      </td>
      <td>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          remove
        </button>
      </td>
    </tr>
  </table>
</div>
`;

exports[`it renders VoicemailTable without data 1`] = `
<div
  className="voicemailContainer"
>
  <h2>
    Voicemails
  </h2>
  <table
    className="voicemailTable"
  >
    <colgroup>
      <col
        className="numberColumn"
      />
      <col />
      <col
        className="durationColumn"
      />
      <col
        className="removeColumn"
      />
    </colgroup>
    <tr>
      <th>
        Number
      </th>
      <th>
        Text
      </th>
      <th>
        Duration
      </th>
      <th />
    </tr>
    <tr>
      <td
        colSpan={3}
      >
        There is no entry
      </td>
    </tr>
  </table>
</div>
`;
